aten/src/ATen/core/ivalue.h

```diff
@@ -321,11 +321,23 @@ struct CAFFE2_API IValue final {
   c10::List<IValue> toList() const &;
   c10::ArrayRef<IValue> toListRef() const;
 
+  // Some template constructors of IValue calls another constructor recursively.
+  // This SNIFEAs the called constructor exists.
   template<class T>
+  using enable_if_recursive_ctor_exists =
```

Comment:
`enable_if_ivalue_constructible` seems like a clearer name for this template. What do you think?

Reply:
I agree. I'll change it.